src/main/java/bio/terra/cli/service/SamService.java

```diff
@@ -550,22 +417,108 @@ static boolean isRetryable(Exception ex) {
   }
 
   /**
-   * Helper method to call {@link
-   * HttpUtils#callAndHandleOneTimeErrorWithRetries(HttpUtils.SupplierWithCheckedException,
-   * Predicate, Predicate, HttpUtils.SupplierWithCheckedException, Predicate)} with the {@link
-   * #isRetryable(Exception)} method as both isRetryable arguments.
+   * Execute a function that includes hitting SAM endpoints. Retry if the function throws an {@link
+   * #isRetryable} exception. If an exception is thrown by the SAM client or the retries, make sure
+   * the HTTP status code and error message are logged.
+   *
+   * @param makeRequest function with no return value
+   * @param errorMsg error message for the the {@link SystemException} that wraps any exceptions
+   *     thrown by the SAM client or the retries
+   */
+  private void callWithRetries(
+      HttpUtils.RunnableWithCheckedException<ApiException> makeRequest, String errorMsg) {
+    handleClientExceptions(
+        () -> HttpUtils.callWithRetries(makeRequest, SamService::isRetryable), errorMsg);
+  }
+
+  /**
+   * Execute a function that includes hitting SAM endpoints. Retry if the function throws an {@link
+   * #isRetryable} exception. If an exception is thrown by the SAM client or the retries, make sure
+   * the HTTP status code and error message are logged.
+   *
+   * @param makeRequest function with a return value
+   * @param errorMsg error message for the the {@link SystemException} that wraps any exceptions
+   *     thrown by the SAM client or the retries
+   */
+  private <T> T callWithRetries(
+      HttpUtils.SupplierWithCheckedException<T, ApiException> makeRequest, String errorMsg) {
+    return handleClientExceptions(
+        () -> HttpUtils.callWithRetries(makeRequest, SamService::isRetryable), errorMsg);
+  }
+
+  /**
+   * Execute a function that includes hitting SAM endpoints. If an exception is thrown by the SAM
+   * client or the retries, make sure the HTTP status code and error message are logged.
+   *
+   * @param makeRequest function with no return value
+   * @param errorMsg error message for the the {@link SystemException} that wraps any exceptions
+   *     thrown by the SAM client or the retries
+   */
+  private void handleClientExceptions(
+      HttpUtils.RunnableWithCheckedException<ApiException> makeRequest, String errorMsg) {
+    handleClientExceptions(
```

Comment:
you could also overload this to take a runnable, but if this is the only place you need it, it's fine.

Reply:
This method is the overload to take a runnable. Or maybe I've misunderstood your comment. Please restate if so.

Reply:
It's fine. I was originally thinking you'd add an overload one level upstream in HttpUtils, but yeah, I read it wrong.